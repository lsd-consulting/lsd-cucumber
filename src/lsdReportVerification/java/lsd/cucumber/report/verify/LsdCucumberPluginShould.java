package lsd.cucumber.report.verify;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static java.nio.file.Paths.get;
import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;

class LsdCucumberPluginShould {

    @Test
    void generateMenu() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/lsd-index.html"));
    }

    @Test
    void generateReportForBasicScenario() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario.html"));
    }

    @Test
    void generateReportForScenarioWithDataTable() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Data_table.html"));
    }

    @Test
    void generateForScenarioOutline() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario_outline.html"));
    }

    private File copyOfFile(String path) throws IOException {
        Path original = get(path);
        Path copy = Path.of(original.getParent().toString(), "copy_" + original.getFileName());
        Files.copy(original, copy, REPLACE_EXISTING);
        return copy.toFile();
    }
}
