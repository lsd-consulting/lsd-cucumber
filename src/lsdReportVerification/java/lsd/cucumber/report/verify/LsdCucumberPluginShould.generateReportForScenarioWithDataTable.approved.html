<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title>data_table</title>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/lsd-consulting/lsd-core@5/src/main/resources/static/style.min.css">
</head>
<body>
<main role="main">
    
    <h1 class="logo">LSD</h1>
    <h1>data_table</h1><article id="1">
        <span class="articleStart"></span>
        <h2 class="success">Scenario with a data table</h2><section class="description success">
            <h3 class="success">Description</h3>
            Given  given2</br>When  when2</br>Then  then2
        </section>
        
        <section class="sequence diagram svg success">
            <h3 class="success">Sequence Diagram</h3>
            <figure><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="160px" preserveAspectRatio="none" style="width:189px;height:160px;background:#FFFFFF;" version="1.1" viewBox="0 0 189 160" width="189px" zoomAndPan="magnify"><defs/><g><line style="stroke:#708090;stroke-width:1.0;stroke-dasharray:5.0,5.0;" x1="45" x2="45" y1="41.4883" y2="120.1094"/><line style="stroke:#708090;stroke-width:1.0;stroke-dasharray:5.0,5.0;" x1="132" x2="132" y1="41.4883" y2="120.1094"/><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#7D8590;stroke-width:1.5;" width="61" x="15" y="10"/><text fill="#8392A2" font-family="sans-serif" font-size="14" lengthAdjust="spacing" textLength="47" x="22" y="30.5352">Source</text><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#7D8590;stroke-width:1.5;" width="61" x="15" y="119.1094"/><text fill="#8392A2" font-family="sans-serif" font-size="14" lengthAdjust="spacing" textLength="47" x="22" y="139.6445">Source</text><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#7D8590;stroke-width:1.5;" width="92" x="86" y="10"/><text fill="#8392A2" font-family="sans-serif" font-size="14" lengthAdjust="spacing" textLength="78" x="93" y="30.5352">Destination</text><rect fill="#FFFFFF" height="30.4883" rx="2.5" ry="2.5" style="stroke:#7D8590;stroke-width:1.5;" width="92" x="86" y="119.1094"/><text fill="#8392A2" font-family="sans-serif" font-size="14" lengthAdjust="spacing" textLength="78" x="93" y="139.6445">Destination</text><polygon fill="#7D8590" points="120,68.7988,130,72.7988,120,76.7988,124,72.7988" style="stroke:#7D8590;stroke-width:1.0;"/><line style="stroke:#7D8590;stroke-width:1.5;" x1="45.5" x2="126" y1="72.7988" y2="72.7988"/><a href="#id1" target="_top" title="label1" xlink:actuate="onRequest" xlink:href="#id1" xlink:show="new" xlink:title="label1" xlink:type="simple"><text fill="#656D76" font-family="sans-serif" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="38" x="52.5" y="68.0566">label1</text></a><polygon fill="#7D8590" points="56.5,98.1094,46.5,102.1094,56.5,106.1094,52.5,102.1094" style="stroke:#7D8590;stroke-width:1.0;"/><line style="stroke:#7D8590;stroke-width:1.5;" x1="50.5" x2="131" y1="102.1094" y2="102.1094"/><a href="#id2" target="_top" title="label2" xlink:actuate="onRequest" xlink:href="#id2" xlink:show="new" xlink:title="label2" xlink:type="simple"><text fill="#656D76" font-family="sans-serif" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="38" x="62.5" y="97.3672">label2</text></a></g></svg></figure>
            <div onclick="location.href='#!';" id="2" class="overlay">
                <div class="popup" onclick="event.stopPropagation();">
                    <h2>PlantUml markup</h2>
                    <span class="copy" title="copy" onclick="copyToClipboard('sequence_uml_code2')"></span>
                    <a class="close" href="#!">&times;</a>
                    <div class="content">
                        <pre><code id="sequence_uml_code2">@startuml

!theme lsd-light from https://cdn.jsdelivr.net/gh/lsd-consulting/lsd-core@5/src/main/resources/static/

hide unlinked




participant Source

participant Destination



Source -&gt; Destination: [[#id1 {label1} label1]]

Destination -&gt; Source: [[#id2 {label2} label2]]


@enduml</code></pre>
                    </div>
                </div>
            </div>
            <sub><a href="#2">[source]</a></sub>
            <sub><a href="#3">[component diagram]</a></sub>
        </section>
        <!-- Popup for component diagram -->
        <div onclick="location.href='#!';" id="3" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>Component Diagram</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre><?xml version="1.0" encoding="us-ascii" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="153px" preserveAspectRatio="none" style="width:137px;height:153px;" version="1.1" viewBox="0 0 137 153" width="137px" zoomAndPan="magnify"><defs/><g><!--entity Source--><g id="elem_Source"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="88" x="27" y="17"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="95" y="22"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="93" y="24"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="93" y="28"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="42" y="51.0752">Source</text></g><!--entity Destination--><g id="elem_Destination"><rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="120" x="11" y="100"/><rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="111" y="105"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="109" y="107"/><rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="109" y="111"/><text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="26" y="134.0752">Destination</text></g><path d="M64.8766,64.063 C64.0451,75.204 64.0411,88.6144 64.8648,99.7771 " fill="none" style="stroke:#000000;stroke-width:1.0;"/><polygon fill="#000000" points="64.8648,99.7771,68.1917,90.5071,64.4968,94.7906,60.2134,91.0958,64.8648,99.7771" style="stroke:#000000;stroke-width:1.0;"/><path d="M77.1352,99.7771 C77.9589,88.6144 77.9549,75.204 77.1234,64.063 " fill="none" style="stroke:#000000;stroke-width:1.0;"/><polygon fill="#000000" points="77.1234,64.063,73.8043,73.3358,77.4955,69.0491,81.7821,72.7403,77.1234,64.063" style="stroke:#000000;stroke-width:1.0;"/></g></svg></pre>
                </div>
                <sub><a href="#uml3">[source]</a></sub>
            </div>
        </div>
        <!-- Popup for component diagram UML via component diagram popup -->
        <div onclick="location.href='#!';" id="uml3" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>PlantUml markup</h2>
                <span class="copy" title="copy" onclick="copyToClipboard('component_uml_code3')"></span>
                <a class="close" href="#3">&times;</a><!--show previous popup on close-->
                <div class="content">
                    <pre><code id="component_uml_code3">@startuml
!pragma layout smetana
!theme plain
skinparam backgroundColor transparent

component Source
component Destination


[Source] --&gt; Destination
[Destination] --&gt; Source


remove @unlinked
@enduml</code></pre>
                </div>
            </div>
        </div>
        <!-- Section containing data for sequence diagram popups --><div onclick="location.href='#!';" id="id1" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label1</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div><div onclick="location.href='#!';" id="id2" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label2</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div>
    </article>
</main>
</body>
<footer>
    <script src="https://cdn.jsdelivr.net/npm/d3@7"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11/build/styles/dark.min.css">
    <script src="https://cdn.jsdelivr.net/gh/lsd-consulting/lsd-core@5/src/main/resources/static/custom.min.js"></script>
    <script>
        addSliderForSvgZoom();
        highlightLifelinesWhenClicked();
    </script>
    <script src="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11/build/highlight.min.js"></script>
    <script>hljs.highlightAll();</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js"></script>
    <script>
        highlightKeywords();
        unMarkSvg();
    </script>
</footer>
</html>
